snips-nlu-ffi/python/snips_nlu_rs/nlu_engine.py

```diff
@@ -19,10 +20,13 @@ def __init__(self, data_path=None, data_zip=None):
         if data_path is None and data_zip is None:
             raise ValueError("Please specify data_path or data_zip")
 
-        if data_path is not None:
-            self._engine = pointer(c_void_p())
+        self._engine = pointer(c_void_p())
+        if os.path.isdir(data_path):
             exit_code = lib.nlu_engine_create_from_dir(
                 data_path.encode("utf-8"), byref(self._engine))
+        else:
+            exit_code = lib.nlu_engine_create_from_file(
+                data_path.encode("utf-8"), byref(self._engine))
```

Comment:
`data_path` can be None here. You must wrap this within a `if data_path is not None:` condition to avoid creating two `pointer(c_void_p())`
 in case where `data_path` is None and `data_zip` is not None